### PR TITLE
Add new Servicegraph visualization.

### PR DESCRIPTION
**Pull request by jeffmendoza**
Adds a new viz under /force/forcegraph.html. New serialization needed
in d3graph.go. Update README.

### CODE REVIEW

File: addons/servicegraph/README.md [hunk 1]
```diff
@@ -1,28 +1,44 @@
...
 
-These services are examples ONLY. This code may change at will, or be removed
-entirely without warning. Taking any dependency on this code is done at your own
-peril.
+## Visualizations
 
-## Services
+- `/force/forcegraph.html` is an interactive
+  [D3.js](https://d3js.org/) visualization. This graph only displays
+  services that are currently receiving traffic.
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply addons/servicegraph/README.md]: Yes, the js code queries the backend with those parameters. The default I chose is `time_horizon=10s&filter_empty=true`, as this leads to good demos of adding services and/or changing route rules.

My idea was to have UI elements to toggle those things, but I don't have those now. I think some js to pass those parameters through would be an easy interim solution.

File: addons/servicegraph/README.md [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
-All endpoints take an optional argument of `time_horizon`, which controls the 
-timespan to consider for graph generation.
+- `/dotgraph` provides a
+  [DOT](https://en.wikipedia.org/wiki/DOT_(graph_description_language))
+  serialization.
 
-All endpoints also take an optional arugment of `filter_empty=true`, which will
-restrict the nodes and edges shown to only those that reflect non-zero traffic
-levels during the specified `time_horizon`.
+- `/d3graph` provides a JSON serialization for D3 visualization.
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply addons/servicegraph/README.md]: I was more comfortable doing the transformation in Go. I think the core Static/Dynamic graph structure is better for internal representation than the one that D3 wants, so I didn't want to change that. The `/graph` endpoint could be removed, as it is not used, but It is trivial to keep it, and someone might find a way to use it.

File: addons/servicegraph/force/forcegraph.css [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    stroke: white;
+    stroke-width: 1px;
+    fill: white;
+}
+
+.node {
+    stroke: black;
+    stroke-width: 1.5px;
+    cursor: move;
+    fill: #466BB0;
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply addons/servicegraph/force/forcegraph.css]: It is Istio blue now, I had it as "lightcoral" before.

File: addons/servicegraph/force/forcegraph.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<head>
+  <title>Istio Service Graph</title>
+  <meta charset="utf-8">
+  <link rel="stylesheet" href="forcegraph.css">
+</head>
+<body>
+  <div id="total">
```
[the remark this answers is not in the document]
Comment (jeffmendoza) [reply addons/servicegraph/force/forcegraph.html]: Will Do.

File: addons/servicegraph/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (douglas-reid) [reply addons/servicegraph/README.md]: ok. fair enough.

File: addons/servicegraph/force/forcegraph.css (hunk 3 above)
[the remark this answers is not in the document]
Comment (douglas-reid) [reply addons/servicegraph/force/forcegraph.css]: ha. ok. i need to learn hex colors ;).

**Comment by jeffmendoza** [comment]
@douglas-reid Comments addressed.

**Comment by ldemailly** [comment]
in the future any chance to move those large js files to contrib/ or use CDN urls ?

**Comment by jeffmendoza** [comment]
No plans at the moment, I think it is better to serve them from the container.

I do think all of addons/ should be moved out to the contrib repo at some point. Right now build/release uses only istio/istio, and it's best to keep the addons versioned and released with the rest of Istio for now.

I imagine the release process will grow to encompass more repos again at some point.

**Pull request merged**